Completed Week 2: Basic portfolio website

carolksun opened the pull request:
I finished creating a static html website to highlight my portfolio, following the instructions outlined in the provided github. My website includes multiple pages, such as an about me, background, photos, and a contact me page.

I have yet to deploy the website since I am waiting for the GCP billing to be set up.

carolksun committed:
Update the README.md file to describe the repo.

carolksun committed:
change text, add picture, generates random fact

carolksun committed:
add image

Commit:
First draft of portfolio website

Write about me and background page
Add photos
Link to Github and Linkedin
Apply gradient background
Change colors to look better

Commit:
Change design of portfolio website

Add multiple pages
Include header clicks
Show logo

Commit:
Fix design issues

Change menu text to white when on page
Change font size
Add line breaks to paragraphs

Commit:
replace photo with smaller version to increase loading speed

razvanm commented on portfolio/src/main/webapp/style.css:
```diff
@@ -1,9 +1,94 @@
+
+html, body {min-height: 100%;}
+
+.container{
```
For consistency there should be a space before `{`. Same in similar places from below.

razvanm commented on portfolio/src/main/webapp/script.js:
```diff
@@ -15,14 +15,44 @@
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function addRandomFact() {
+  const facts =
+      ['I\'ve lived in LA my whole life', 'I have surpassed level 3000 on Candy Crush Saga', 'I have an older sister', 'My favorite fruit is watermelon'];
```
Having each string on a separate line would make this easier to read. You can also avoid having the `\'` by using double quotes (this file seems to use both single quotes and double quotes for strings).

razvanm commented on portfolio/src/main/webapp/script.js:
```diff
@@ -15,14 +15,44 @@
+  factContainer.innerText = fact;
 }
+
+function setupTabs (){
```
Space between `()` and `{`.

Would be good to add a function comment to indicate what this function is doing.

carolksun replied:
I deleted the setupTabs and eventListener because it was part of a previous feature that I no longer use. But I'll take note of commenting functions!

razvanm commented on portfolio/src/main/webapp/script.js:
```diff
@@ -15,14 +15,44 @@
+  setupTabs();
+
+  document.querySelectorAll(".tabs").forEach(tabsContainer => {
+    tabsContainer.querySelector(".tabs__sidebar .tabs__button").click();
```
It's not obvious why this code is not part of the `setupTabs`. A function comment would help.

razvanm commented on portfolio/src/main/webapp/index.html:
```diff
@@ -1,18 +1,36 @@
+          <p>Hi! I'm Carol Sun! Here is a picture of me.</p>
+          <img src="images/Profile Pic.jpg" alt="Me" height="300" class = "center"/>
+            <p>I am currently a STEP intern and a rising junior at Caltech. I live in the 
+            Los Angeles area and I like to cook and garden in my free time!</p>
```
Is this indentation intentional?

carolksun replied:
oops! I fixed the indentation

razvanm commented on portfolio/src/main/webapp/contact.html:
```diff
@@ -0,0 +1,28 @@
+    <script src="script.js"></script>
+  </head>
+  <body>
+    <div class = "container">
+      <div class = "menu">
```
It looks unusual to have spaces around `=`. It's good to be consistent.

carolksun committed:
Add space before { and make spacing consistent

carolksun committed:
Make each sentence in array a separate line

Change the ' to "

carolksun committed:
Remove code regarding tabs

The tabs was part of a previous draft of my website and I no longer use the tab features.

carolksun committed:
Remove spaces around =

carolksun committed:
Fix indentation

carolksun commented:
I fixed all the comments you left! In my next PR, I'll try to be more careful with my spacing.

yinggao commented on portfolio/src/main/webapp/photos.html:
```diff
@@ -0,0 +1,32 @@
+    <script src="script.js"></script>
+  </head>
+  <body>
+    <div class = "container">
+      <div class = "menu">
```
Remove the space between '=' to be consistent.

yinggao commented on portfolio/src/main/webapp/index.html:
```diff
@@ -1,18 +1,36 @@
-      <button onclick="addRandomGreeting()">Hello</button>
-      <div id="greeting-container"></div>
+    <div class = "container">
+      <div class = "menu">
```
Ditto. Review the spaces around '=' to be consistent.

yinggao commented on portfolio/src/main/webapp/photos.html:
```diff
@@ -0,0 +1,32 @@
+            <li class = "logo"> <a href="index.html"> <img src="images/logo.jpg"> </a> </li>
+            <li><a href="index.html"> About Me</a></li>
+            <li> <a href="background.html"> Background</a></li>
+            <li class = "currentLink"> <a href="photos.html" style="color:white"> Photos </a> </li>
```
Same here. Remove the space between '='.

yinggao commented on portfolio/src/main/webapp/background.html:
```diff
@@ -0,0 +1,29 @@
+        <ul>
+            <li class = "logo"> <a href="index.html"> <img src="images/logo.jpg"> </a> </li>
+            <li><a href="index.html"> About Me</a></li>
+            <li class = "currentLink"> <a href="background.html" style="color:white"> Background</a></li>
```
Same here, remove the spaces around '='.

razvanm commented on portfolio/src/main/webapp/background.html:
```diff
@@ -0,0 +1,29 @@
+      </div>
+    </div>
+    <h1> Background</h1>
+    <p> I am a computer science major at Caltech, expecting to graduate in June 2022. <br> So far, some of the theoretical CS classes I have taken are algorithms, decidability and tractability, and learning systems. <br> Some of the practical CS classes I've taken are data structures, software design, and machine learning. <br> I'm excited to be taking some more in-depth CS classes in my next two ...
+    </p>
```
A slightly nicer formatting:

```
  <p>
    I am a computer science major [...] <br>
    So far [...]
  </p>
```

razvanm commented on portfolio/src/main/webapp/background.html:
```diff
@@ -0,0 +1,29 @@
+      <div class = "menu">
+        <ul>
+            <li class = "logo"> <a href="index.html"> <img src="images/logo.jpg"> </a> </li>
+            <li><a href="index.html"> About Me</a></li>
```
Would be good to be consistent with spacing inside the `<a>` element:

- if there is not space before `</a>` then there should be no space after `<a>` either 
- if there a space after `<a>` then there should be a space before `</a>` too.

razvanm commented on portfolio/src/main/webapp/contact.html:
```diff
@@ -0,0 +1,28 @@
+      <div class="menu">
+        <ul>
+          <li class="logo"> <a href="index.html"> <img src="images/logo.jpg"> </a> </li>
+          <li><a href="index.html"> About Me</a></li>
```
The other `<li>` elements have a space inside them while this one doesn't.

carolksun committed:
Remove spaces around =

carolksun committed:
Add new line after line break

carolksun committed:
Add space between <li> and <a>

carolksun committed:
Remove spaces around =

carolksun committed:
Remove spaces around = and make spacing symmetric

carolksun committed:
Add spaces around tags

carolksun committed:
Add spaces around tags

carolksun committed:
Add space between </a> and </li>

carolksun committed:
Add space around tags

carolksun committed:
Add space between </a> and </li>

razvanm commented on portfolio/src/main/webapp/style.css:
```diff
@@ -1,9 +1,95 @@
+
```
Leaving empty lines at the top a file is unusual.

razvanm commented on portfolio/src/main/webapp/background.html:
```diff
@@ -0,0 +1,35 @@
+      </div>
+    </div>
+    <h1> Background </h1>
+    <p> I am a computer science major at Caltech, expecting to graduate in June 2022. <br> 
```
This would look better if `<p>` is alone on the line. That would also make it consistent with the fact that `</p>` is alone on its line.

razvanm commented on portfolio/src/main/webapp/background.html:
```diff
@@ -0,0 +1,35 @@
+      I'm excited to be taking some more in-depth CS classes in my next two years at Caltech, including operating systems, information theory, and databases. 
+    </p>
+    <p>
+    I've been lucky to have had a few internships before and during college, <br> 
+      including a summer research project at the National Institutes of Health studying Multiple Sclerosis, <br> 
```
The `I've` should be two spaces to the right to be align with the line from below.

razvanm commented on portfolio/src/main/webapp/contact.html:
```diff
@@ -0,0 +1,28 @@
+      </div>
+    </div>
+    <h1> Contact Me </h1>
+    <p> I have a more visually appealing website located <a href="https://carolksun.github.io"> here </a>.
+    </p>
+    <p> Feel free to check out my <a href="https://www.linkedin.com/in/carolkailesun"> Linkedin </a> and <a href="http://www.github.com/carolksun"> Github </a>!
+    </p>
```
Same observation about formatting `<p>`s as in the background.html file.

carolksun committed:
Fix indentation

carolksun committed:
Fix indentation for <p> tag

carolksun committed:
Remove empty line at top

razvanm reviewed (approved):
Note that the PR title is not suitable for the commit subject. I would also suggest wrapping the body for the commit message to 80-columns because it makes it look better when using command line tools.
